frontends/pytorch/csrc/builder/class_annotator.h

```diff
@@ -0,0 +1,125 @@
+//===- class_annotations.h --------------------------------------*- C++ -*-===//
+//
+// This file is licensed under a pytorch-style license
+// See frontends/pytorch/LICENSE for license information.
+//
+//===----------------------------------------------------------------------===//
+// Utilities for annotating Torch `c10::ClassType`
+//
+// We cannot intrusively add metadata to the ClassType, so we instead
+// keep a parallel data structure.
+//
+// An annotation injects extra knowledge about the program which is not
+// otherwise deducible. Thus, it is important that all annotations have a safe
+// "no extra knowledge" state.
+//
+// Annotations should not be thought of at the MLIR level. They should express
+// information at the level of the user-observable program semantics independent
+// of implementation.
+//===----------------------------------------------------------------------===//
+
+#ifndef NPCOMP_FRONTENDS_PYTORCH_CSRC_CLASS_ANNOTATOR_H
+#define NPCOMP_FRONTENDS_PYTORCH_CSRC_CLASS_ANNOTATOR_H
+
+#include "../pybind.h"
+
+namespace torch_mlir {
+
+// An annotation on a class's attribute (corresponds to a c10::ClassAttribute).
+struct AttributeAnnotation {
+  // Whether external access to this attribute is allowed.
+  // The default "no knowledge" state of the program is that all attributes
+  // can be externally accessed.
+  bool isExported = true;
+
+  std::string toString(const std::string &name);
+};
+
+// An annotation on a class's method (corresponds to a torch::jit::Function).
+struct MethodAnnotation {
+  // Whether external calls to this method are allowed.
+  // The default "no knowledge" state of the program is that all methods
+  // can be externally called.
+  bool isExported = true;
```

Comment:
I do wonder whether this is the right long term choice... In my experience, there can be quite a bit of methods that are not top of mind for the user (due to using utilities, etc). Fine to see how this works out in practice to decide (just making a note to think about it later).

Reply:
In that case users can start with a call to `exportNone` (as we currently do everywhere we use this). This a "laws of physics of compilers" sort of thing. We need to be conservative -- there's really no other choice in terms of modeling this.